### PR TITLE
fix #2740 reintroduce show_term_document_count_error which went blata…

### LINKED ISSUE #2740: ShowTermDocumentCountError in 5.x?
Hey everyone,

I'm not sure whether this is a bug, or if it's a feature request, but please bear with me.

I'm running Elasticsearch 5.3.0, and have NEST and Elasticsearch-Net 5.3.0 as well.

I'm using the terms aggregation, and I just came across the `show_term_doc_count_error` setting.

The only reference I can find in this repo to this setting is in the list of 5.0 breaking changes, where it's listed as removed (specifically, in its C# form, `ShowTermDocumentCountError`). Here's that document: https://github.com/elastic/elasticsearch-net/blob/781430ec6793153a87f8e38ae93448b9db9fe6e6/docs/5.0-breaking-changes/nest-breaking-changes.asciidoc

However, I'm able to use it in queries against my cluster, again running >5.0, and I don't see anything in the Elasticsearch-side documentation that hints at this being deprecated. I see the giant warning that it's experimental and *may* be removed, but nothing to say it has been already.

Is there something I'm missing here? Was this setting properly removed from the SDK?

Thanks,
Matthew

### PR DESCRIPTION
**Pull request by Mpdreamz**
…ntly missing in 5.x

### CODE REVIEW

File: src/Nest/Aggregations/AggregateJsonConverter.cs [hunk 1]
```diff
@@ -73,20 +74,20 @@ static AggregateJsonConverter()
...
 				.ToArray();
 
 			var allKeys = string.Join(", ", AllReservedAggregationNames);
 			UsingReservedAggNameFormat =
 				"'{0}' is one of the reserved aggregation keywords"
-			    + " we use a heuristics based response parser and using these reserved keywords"
-			    + " could throw its heuritics off course. We are working on a solution in elasticsearch itself to make"
-			    + " the response parsable. For now these are all the reserved keywords: "
+				+ " we use a heuristics based response parser and using these reserved keywords"
+				+ " could throw its heuritics off course. We are working on a solution in elasticsearch itself to make"
```
Comment (russcam) [inline src/Nest/Aggregations/AggregateJsonConverter.cs]: minor nitpick: `heuristics` spelling

File: src/Nest/Aggregations/AggregateJsonConverter.cs [hunk 2]
```diff
@@ -73,20 +74,20 @@ static AggregateJsonConverter()
...
+				+ " the response parsable. For now these are all the reserved keywords: "
```
Comment (russcam) [inline src/Nest/Aggregations/AggregateJsonConverter.cs]: minor nitpick: `parseable` spelling

**Comment by Mpdreamz** [comment]
@russcam for a moment I installed resharper and rider took some settings from it. We need to find a moment to sort resharper settings out. Been putting it off for too long now :smile:

Next monday go over it?

File: src/Nest/Aggregations/AggregateJsonConverter.cs (hunk 1 above)
Comment (Mpdreamz) [inline src/Nest/Aggregations/AggregateJsonConverter.cs]: 👍

File: src/Nest/Aggregations/AggregateJsonConverter.cs (hunk 2 above)
Comment (Mpdreamz) [inline src/Nest/Aggregations/AggregateJsonConverter.cs]: 👍

**Comment by Mpdreamz** [comment]
Updated my horrible spelling :smile:

**Comment by Mpdreamz** [comment]
ported to `master`

**Pull request merged**